hazelcast/src/main/java/com/hazelcast/query/impl/AndResultSet.java

```diff
@@ -161,6 +153,28 @@ public void remove() {
 
     @Override
     public int size() {
-        return setSmallest.size();
+        if (cachedSize == SIZE_UNINITIALIZED) {
+            int calculatedSize = 0;
+            for (Iterator<QueryableEntry> it = iterator(); it.hasNext(); it.next()) {
+                calculatedSize++;
+            }
+            cachedSize = calculatedSize;
+        }
+        return cachedSize;
     }
+
+    /**
+     * @return returns estimated size without calculating the full result set in full-result scan.
+     */
+    public int estimatedSize() {
```

Comment:
Good question, but the answer is `no` since the `estimatedSize()` won't cache anything.
It uses the `cachedSize` if it is initialized (since it delivers the most accurate result), but if there's no `cachedSize` it will return an estimated size without caching it anywhere.